lib/galaxy_test/base/populators.py

```diff
@@ -1674,19 +1675,25 @@ def _create_collection(self, payload: dict) -> Response:
 
 def load_data_dict(history_id, test_data, dataset_populator, dataset_collection_populator):
     """Load a dictionary as inputs to a workflow (test data focused)."""
-
     def open_test_data(test_dict, mode="rb"):
+        """
+        load test data given in test_dict["value"]
+        - if its a file name: a handle to the file
+        - otherwise: a handle to the raw string data
+        """
         test_data_resolver = TestDataResolver()
         filename = test_data_resolver.get_filename(test_dict["value"])
-        return open(filename, mode)
+        if filename is not None:
+            return open(filename, mode)
+        else:
+            return StringIO(test_dict["value"])
```

Comment:
This is an optional suggestion totally biased by having done this in https://github.com/galaxyproject/galaxy/pull/13110/files, but I like the distinction CWL makes here. If `content` is provided it is used as a file literal. So instead of deciding this based on whether value is in the test data dir, you could restore the original version and use something like
```python
            input_type = value["type"]
            if input_type == "File":
                if 'content' in value:
                    content = value.get('content')
                else:
                    content = open_test_data(value)
```
on line 1738, together with changing `value` to `content` in the test_data

```suggestion
        return open(filename, mode)
```

Reply:
Yeah - we cannot just rework what value here means I think given that we want to work on how value is resolved.